File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day22GridComputingTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day22GridComputingTest  extends FlatSpec with Matchers {
  "2016 Day 22 - Input" should "calculate answers" in {
    val dp = new Day22GridComputing("/twenty_sixteen/Day22-GridComputing-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day22GridComputing("/twenty_sixteen/Day22-GridComputing-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}

File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day22GridComputingTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day22GridComputingTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day22GridComputing("/twenty_sixteen/Day22-GridComputing-input.txt");

        assertEquals(1003, dp.getPart1Answer().intValue());
        assertEquals(192, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day22GridComputing("/twenty_sixteen/Day22-GridComputing-example#1.txt");

        assertEquals(7, dp.getPart1Answer().intValue());
    }
}